File: python/algo/202607/algo0554.py
from collections import defaultdict
from typing import List


class Solution:
    def leastBricks(self, wall: List[List[int]]) -> int:
        n, mx = len(wall), 0
        board = defaultdict(int)
        for row in wall:
            m, right = len(row), 0
            for i in range(m-1):
                right += row[i]
                board[right] += 1
                if board[right] > mx: mx = board[right]

        return n - mx

File: cplus/leetcode/202607/algo0554.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int leastBricks(vector<vector<int>>& wall) {
        int n = wall.size(), mx = 0;
        unordered_map<long long, int> slot;
        for (auto row: wall) {
            int m = row.size(); long long right = 0;
            for (int i=0; i<m-1; i++) {
                right += row[i];
                slot[right] ++;
                mx = max(slot[right], mx);
            }
        }
        
        return n - mx;
    }
};